app/components/primer/select_menu_component.rb

```diff
@@ -0,0 +1,323 @@
+# frozen_string_literal: true
+
+module Primer
+  class SelectMenuComponent < Primer::Component
+    include ViewComponent::Slotable
+
+    LIST_BORDER_CLASSES = {
+      all: nil,
+      omit_top: "border-top-0",
+      none: "SelectMenu-list--borderless",
+    }.freeze
+    DEFAULT_LIST_BORDER_CLASS = :all
+    DEFAULT_LOADING = false
+    DEFAULT_BLANKSLATE = false
+    DEFAULT_ALIGN_RIGHT = false
+
+    with_slot :header, class_name: "Header"
+    with_slot :item, class_name: "Item", collection: true
+    with_slot :tab, class_name: "Tab", collection: true
+    with_slot :filter, class_name: "Filter"
+    with_slot :footer, class_name: "Footer"
+
+    attr_reader :message
+
+    def initialize(
+      align_right: DEFAULT_ALIGN_RIGHT,
+      loading: DEFAULT_LOADING,
+      blankslate: DEFAULT_BLANKSLATE,
+      list_border: DEFAULT_LIST_BORDER_CLASS,
+      list_role: nil,
+      message: nil,
+      **kwargs
+    )
+      @align_right = fetch_or_fallback([true, false], align_right, DEFAULT_ALIGN_RIGHT)
+      @loading = fetch_or_fallback([true, false], loading, DEFAULT_LOADING)
+      @blankslate = fetch_or_fallback([true, false], blankslate, DEFAULT_BLANKSLATE)
+      @list_border = fetch_or_fallback(LIST_BORDER_CLASSES.keys, list_border,
+        DEFAULT_LIST_BORDER_CLASS)
+      @list_role = list_role
+      @message = message
+      @kwargs = kwargs
+      @kwargs[:tag] ||= :div
+      @kwargs[:role] ||= "menu" if @kwargs[:tag] == :"details-menu"
+    end
+
+    def render?
+      items.any? || tabs.any? || content.present? || message.present?
+    end
+
+    private
+
+    def loading?
+      @loading
+    end
+
+    def blankslate?
+      @blankslate
+    end
+
+    class Tab < Primer::Slot
+      DEFAULT_SELECTED = false
+
+      attr_reader :selected
+
+      def initialize(selected: DEFAULT_SELECTED, **kwargs)
+        @selected = fetch_or_fallback([true, false], selected, DEFAULT_SELECTED)
+        @kwargs = kwargs
+        @kwargs[:tag] = :button
+        @kwargs[:classes] = class_names(
+          "SelectMenu-tab",
+          kwargs[:classes],
+        )
+        @kwargs[:"aria-selected"] = "true" if @selected
```

Comment:
pretty sure `"aria-selected"` doesn't need to be a symbol

Reply:
I think I got a test error when it was a string.